components/eventing-controller/testing/subscriber.go

```diff
@@ -78,9 +79,28 @@ func NewSubscriber() *Subscriber {
 			return
 		}
 	})
+	return mux
+}
 
+func NewSubscriber() *Subscriber {
+	mux := getServeMux()
 	server := httptest.NewServer(mux)
+	return &Subscriber{
+		server:           server,
+		SinkURL:          fmt.Sprintf("%s%s", server.URL, storeEndpoint),
+		checkURL:         fmt.Sprintf("%s%s", server.URL, checkEndpoint),
+		InternalErrorURL: fmt.Sprintf("%s%s", server.URL, internalErrorEndpoint),
+		checkRetriesURL:  fmt.Sprintf("%s%s", server.URL, checkRetriesEndpoint),
+	}
+}
 
+func NewSubscriberWithListener(listener net.Listener) *Subscriber {
```

Comment:
How about we have only one Subscriber factory function which takes variadic options?

For example:

```golang
type SubscriberOption func(*Subscriber)
func NewSubscriber(opts ...SubscriberOption) *Subscriber
func WithListener(listener net.Listener) SubscriberOption
```

Reply:
Reworked as mentioned and discussed! Thanks for pointing this out!